### PR TITLE
Efficient swc build

### PR DESCRIPTION
**Pull request by mgeplf**
New SWC parser:
 * Parsing an SWC goes from 1.01ms to 0.700ms
 * Supports locales

### CODE REVIEW

File: src/readers/morphologySWC.cpp [hunk 1]
```diff
@@ -53,397 +162,374 @@ std::vector<unsigned int> gatherLineNumbers(const std::vector<SWCSample>& sample
...
-       and update the parent ID of the new section
-    */
-    void _processSectionStart(mut::Morphology& morph, const SWCSample& sample) {
-        Property::PointLevel properties;
+        const SWCSample* sample = &samples_.at(id);
 
-        uint32_t id = 0;
+        // create duplicate point if needed
+        if (!is_root && sample->point != start_point /*|| sample->diameter != start_diameter */) {
+        //if (!(is_root && sample->type == SECTION_SOMA) && sample->point != start_point /*|| sample->diameter != start_diameter */) {
```
Comment (eleftherioszisis) [inline src/readers/morphologySWC.cpp:470]: Is this intentionally commented out?

File: src/readers/utils.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+#endif
+
+StringToNumber::StringToNumber()
+#if defined(_WIN32) || defined(__WIN32__) || defined(WIN32)
+    : locale(_create_locale(LC_ALL, "C")) {
+}
+#else
+    : locale(newlocale(LC_NUMERIC_MASK, "POSIX", nullptr)) {
+}
+#endif
```
Comment (eleftherioszisis) [inline src/readers/utils.cpp:31]: Same here. It's not clear to me why these are configured this way.

**Review by eleftherioszisis — APPROVED** [review]

File: src/readers/utils.cpp (hunk 2 above)
Comment (mgeplf) [reply src/readers/utils.cpp]: done

File: src/readers/morphologySWC.cpp (hunk 1 above)
Comment (mgeplf) [reply src/readers/morphologySWC.cpp]: fixed, thanks.

**Pull request merged**